src/Components/WebAssembly/WebAssembly/src/HotReload/WebAssemblyHotReload.cs

```diff
@@ -50,14 +45,19 @@ internal static async Task InitializeAsync()
     [JSInvokable(nameof(ApplyHotReloadDelta))]
     public static void ApplyHotReloadDelta(string moduleIdString, byte[] metadataDelta, byte[] ilDelta, byte[] pdbBytes)
     {
+        // Analyzer has a bug where it doesn't handle ConditionalAttribute: https://github.com/dotnet/roslyn/issues/63464
+#pragma warning disable IDE0200 // Remove unnecessary lambda expression
+        Interlocked.CompareExchange(ref _hotReloadAgent, new HotReloadAgent(m => Debug.WriteLine(m)), null);
```

Comment:
Why does the initialization in the `InitializeAsync` method is not enough? Does the value gets discarded through some other codepath?

Reply:
As the InitializeAsync is an async method what was happening is that in a race condition before _hotReloadAgent is initialized on InitializeAsync it was trying to be used on ApplyHotReloadDelta method and throwing the NRE exception.

Reply:
That sounds like a bug in the consuming code then, rather than here, isn't it? So the consumder then should just make sure they wait for the initialization before trying to apply changes.

Reply:
I'm not sure if it's possivel to check if InitializeAsync is done before calling the ApplyHotReloadDelta as it's called by dotnet watch.

![image](https://user-images.githubusercontent.com/4503299/221947273-932fe8be-96d8-4a01-b69d-798c29ac8c44.png)

Reply:
It is possible that there is an issue with initialization. I am not sure what exactly the initialization sequence for WASM is supposed to do. It'd be great if anyone knows and can review the code paths.

Reply:
Here is where InitializeAsync is called:
https://github.com/dotnet/aspnetcore/blob/6647d80ab886b8616e5b34812853e00358c93ce5/src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHost.cs#L141

And here is where it's calling the applyHotReload function on JS from dotnet watch:
https://github.com/dotnet/sdk/blob/fbc089ea96f0187c31e9ae176278334c8d5defc6/src/BuiltInTools/BrowserRefresh/BlazorHotReload.js#L16

Which will call the ApplyHotReloadDelta on C# side:
https://github.com/dotnet/aspnetcore/blob/6647d80ab886b8616e5b34812853e00358c93ce5/src/Components/Web.JS/src/Boot.WebAssembly.ts#L48

I think there is no way to guarantee that InitializeAsync is done before calling the ApplyHotReloadDelta, I guess that this is the correct fix.